cdc/scheduler/internal/v3/replication/replication_set.go

```diff
@@ -90,16 +90,49 @@ func (r ReplicationSetState) MarshalJSON() ([]byte, error) {
 	return json.Marshal(r.String())
 }
 
+// CaptureRole is the role of a capture.
+type CaptureRole int
+
+const (
+	// CaptureRolePrimary primary role.
+	CaptureRolePrimary = 1
+	// CaptureRoleSecondary secondary role.
+	CaptureRoleSecondary = 2
+	// CaptureRoleUndetermined means that we don't known its state, it may be
+	// replicating, stopping or stopped.
+	CaptureRoleUndetermined = 3
+)
+
+func (r CaptureRole) String() string {
+	switch r {
+	case CaptureRolePrimary:
+		return "Primary"
+	case CaptureRoleSecondary:
+		return "Secondary"
+	case CaptureRoleUndetermined:
+		return "Undetermined"
+	default:
+		return fmt.Sprintf("Unknown %d", r)
+	}
+}
+
+// MarshalJSON returns r as the JSON encoding of CaptureRole.
+// Only used for pretty print in zap log.
+func (r CaptureRole) MarshalJSON() ([]byte, error) {
+	return json.Marshal(r.String())
+}
+
 // ReplicationSet is a state machine that manages replication states.
 type ReplicationSet struct { //nolint:revive
 	Changefeed model.ChangeFeedID
 	TableID    model.TableID
 	State      ReplicationSetState
-	// Primary set indicate that the table is replicating on this capture
+	// Primary is the capture ID that is currently replicating the table.
 	Primary model.CaptureID
-	// Secondary set indicate that the table is preparing on this capture
-	Secondary  model.CaptureID
-	Captures   map[model.CaptureID]struct{}
+	// Captures is a map of captures that has the table replica.
+	// NB: Invariant, 1) at most one primary, 2) primary capture must be in
+	//     CaptureRolePrimary.
+	Captures   map[model.CaptureID]CaptureRole
```

Comment:
I'm quite curious, Why we have `Captures` field, which only have 2 items? 

Why not just directly use `Primary` and `Secondary` fields ?

Reply:
I'm planning to support adding more replicas to tables, so TiCDC can handle unplanned shutdown.